tasks/go.py

```diff
@@ -485,13 +485,14 @@ def check_mod_tidy(ctx, test_folder="testmodule"):
         if os.path.isfile(os.path.join(ctx.cwd, "main")):
             os.remove(os.path.join(ctx.cwd, "main"))
 
-    if errors_found:
-        message = "\nErrors found:\n" + "\n".join("  - " + error for error in errors_found)
-        message += "\n\nRun 'inv tidy-all' to fix 'out of sync' errors."
-        raise Exit(message=message)
-
-    # delete test_folder to avoid FileExistsError while running this task again
-    ctx.run(f"rm -rf ./{test_folder}")
+    try:
```

Comment:
Maybe move this up till the line the directory is created (`generate_dummy_package`).

Reply:
Or even cooler if you want to look into it, we could implement a `with` block like:
```
with generate_dummy_package(ctx, test_folder):
   ...
```
That deletes the folder at the end regardless of exceptions.